guest-agent/VfioService/VfioService/MouseHook.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace VfioService
+{
+    public class MouseHook
+    {
+        private const int WH_MOUSE_LL = 14;
+        private const int WM_MOUSEMOVE = 0x0200;
+
+        private LowLevelMouseProc callback;
+        private IntPtr hookId = IntPtr.Zero;
+        private ClientManager ClientManager;
+
+        public MouseHook(ClientManager clientManager)
+        {
+            ClientManager = clientManager;
+            callback = HookCallback;
+            SetProcessDpiAwareness(ProcessDPIAwareness.ProcessPerMonitorDPIAware);
+            Process curProcess = Process.GetCurrentProcess();
+            ProcessModule curModule = curProcess.MainModule;
+            hookId = SetWindowsHookEx(WH_MOUSE_LL, callback, GetModuleHandle(curModule.ModuleName), 0);
+        }
+
+        ~MouseHook()
```

Comment:
Sounds like it is way easier to just remove the destructor. That's what I'll do then.